src/material-table.js

```diff
@@ -182,42 +182,23 @@ export default class MaterialTable extends React.Component {
 
     calculatedProps.actions = [...(calculatedProps.actions || [])];
 
-    if (calculatedProps.options.selection)
-      calculatedProps.actions = calculatedProps.actions
-        .filter((a) => a)
-        .map((action) => {
-          if (
-            action.position === "auto" ||
-            action.isFreeAction === false ||
-            (action.position === undefined && action.isFreeAction === undefined)
-          )
-            if (typeof action === "function")
-              return { action: action, position: "toolbarOnSelect" };
-            else return { ...action, position: "toolbarOnSelect" };
-          else if (action.isFreeAction)
-            if (typeof action === "function")
```

Comment:
removed because never true
when action is function it always stops at the first if
```
(action.position === undefined && action.isFreeAction === undefined)
```